src/preload.py

```diff
@@ -608,13 +608,14 @@ def write_resin_device_pinning(app_data, output):
             ),
         )
 
-
 def write_apps_json(data, output):
-    """Writes data dict to output as json"""
+    """Updates or creates preloaded state"""
+    if os.path.isfile(output):
+        with open(output, "r") as f:
+            data['apps'].update(json.load(f)['apps'])
```

Comment:
What if the image file had already been preloaded, e.g. if the `balena preload` is executed a second time for the same image file? The earlier execution would take precedence, right? It should be the other way around, the latter execution taking precedence.

Actually, I think it is preferable to forbid executing `balena preload` on the same image file multiple times, because I think users assume that subsequent executions will clean up / undo previous executions, when in fact it doesn't and produces a bloated, unnecessarily large image file. If we could reliably detect that `balena preload` was executed before for this same file (preferably by inspecting the image file itself), I think this would be best. We could print an error message along the lines of:
> Error: this image file has been preloaded before.
> Please use a fresh, not-previously-preloaded image file.

Reply:
@pdcastro I was actually thinking about a multi-app future where users would  run preload several times to load different apps, that's why I implemented it that way.

Reply:
I am distinguishing supervisor from user apps by reading the image names and comparing with the current state. It's ugly, but in future when the supervisor becomes part of the hostOS it can be distinguished with the `isHost` attribute with the rest of the hostOS.

I will raise an exception if trying to re-preload as it seems a better solution.